src/service/acl/data.rs

```diff
@@ -0,0 +1,39 @@
+use std::collections::HashSet;
+
+use clap::ValueEnum;
+use serde::{Deserialize, Serialize};
+use url::Host;
+
+pub trait Data: Send + Sync {
+    /// check if given host exists in Acls, if so return it
+    fn check_acl(&self, host: &Host<String>) -> crate::Result<Option<AclMode>>;
+
+    /// add a given Acl entry to the database
+    fn add_acl(&self, acl: AclDatabaseEntry) -> crate::Result<()>;
+    /// remove a given Acl entry from the database
+    fn remove_acl(&self, host: Host<String>) -> crate::Result<()>;
+
+    /// list all acls
+    fn get_all_acls(&self) -> HashSet<AclDatabaseEntry>;
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, Hash, Eq, PartialEq, ValueEnum)]
```

Comment:
Do we need all these traits implemented? Seems like a lot

Reply:
as far as I know they are needed, hash,eq and partial eq is for hashset, ValueEnum is for clippy idk tho we can remove a bit maybe? clippy doesn't seem to think it's an issue

Reply:
This was just an observation rather than clippy.

Reply:
they are all needed

Reply:
nvm gimme a sec